spec/models/reports/patient_state_spec.rb

```diff
@@ -369,16 +550,28 @@ def patient_states(patient, from: nil, to: nil)
             # off a join with the reporting_months view, which uses now() and so will always have records up until
             # the actual current time.  Timecop only impacts Ruby, and not the actual system time.
             expect(patient_states(patient, from: two_years_ago, to: now).pluck(:months_since_registration)).to eq((0...24).to_a)
+            expect(patient_states(patient, from: two_years_ago, to: now).pluck(:quarters_since_registration))
+              .to eq([0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 5, 5, 6, 6, 6, 7, 7, 7, 8, 8])
 
             expect(patient_states(patient, to: ten_months_ago).pluck(:months_since_visit)).to all(be_nil)
             expect(patient_states(patient, from: ten_months_ago, to: eight_months_ago).pluck(:months_since_visit)).to eq((0..1).to_a)
             expect(patient_states(patient, from: eight_months_ago, to: five_months_ago).pluck(:months_since_visit)).to eq((0..2).to_a)
             expect(patient_states(patient, from: five_months_ago, to: now).pluck(:months_since_visit)).to eq((0...5).to_a)
 
+            expect(patient_states(patient, to: ten_months_ago).pluck(:quarters_since_visit)).to all(be_nil)
+            expect(patient_states(patient, from: ten_months_ago, to: june_2021[:now] - 3.months).pluck(:quarters_since_visit)).to all eq 0
+            expect(patient_states(patient, from: june_2021[:now] - 2.months, to: now).pluck(:quarters_since_visit)).to all eq 1
+
             expect(patient_states(patient, to: ten_months_ago).pluck(:months_since_bp)).to all(be_nil)
             expect(patient_states(patient, from: ten_months_ago, to: five_months_ago).pluck(:months_since_bp)).to eq((0..4).to_a)
             expect(patient_states(patient, from: five_months_ago, to: now).pluck(:months_since_bp)).to eq((0...5).to_a)
 
+            expect(patient_states(patient, to: ten_months_ago).pluck(:quarters_since_bp)).to all(be_nil)
+            expect(patient_states(patient, from: ten_months_ago, to: eight_months_ago).pluck(:quarters_since_bp)).to all eq 0
+            expect(patient_states(patient, from: eight_months_ago, to: five_months_ago).pluck(:quarters_since_bp)).to all eq 1
+            expect(patient_states(patient, from: five_months_ago, to: two_months_ago).pluck(:quarters_since_bp)).to all eq 0
+            expect(patient_states(patient, from: two_months_ago, to: now).pluck(:quarters_since_bp)).to all eq 1
+
```

Comment:
If we change the age calculation, I'd love to see timeline specs for age as well.